tests/test_read_data.py

```diff
@@ -0,0 +1,396 @@
+                reason="filename column inaccessible with pandas backend and parquet"
+            ),
+        ),
+    ],
+)
+def test_read_data_fpp_add_filename(
+    mock_multiple_jsonl_files, mock_multiple_parquet_files, backend, file_type
+):
+    input_files = (
+        mock_multiple_jsonl_files
+        if file_type == "jsonl"
+        else mock_multiple_parquet_files
+    )
+
+    df = read_data_fpp(
+        input_files=input_files,
+        backend=backend,
+        file_type=file_type,
+        files_per_partition=NUM_FILES,
+        add_filename=True,
+        input_meta=None,
+        columns=None,
+    )
+
+    assert set(df.columns) == {"filename", "id", "text"}
+    file_names = df["filename"].unique().compute()
+    if backend == "cudf":
+        file_names = file_names.to_pandas()
+
+    assert len(file_names) == NUM_FILES
+    assert set(file_names.values) == {
+        f"test_{file_id}.{file_type}" for file_id in range(NUM_FILES)
+    }
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        "pandas",
+        pytest.param("cudf", marks=pytest.mark.gpu),
+    ],
+)
+@pytest.mark.parametrize(
+    "file_type,add_filename,function_name",
+    [
+        *[("jsonl", True, func) for func in ["read_data_blocksize", "read_data_fpp"]],
+        *[("jsonl", False, func) for func in ["read_data_blocksize", "read_data_fpp"]],
+        *[
+            ("parquet", False, func)
+            for func in ["read_data_blocksize", "read_data_fpp"]
+        ],
+        *[("parquet", True, "read_data_fpp")],
+    ],
+)
+@pytest.mark.parametrize(
+    "cols_to_select", [None, ["id"], ["text", "id"], ["id", "text"]]
+)
+def test_read_data_select_columns(
+    mock_multiple_jsonl_files,
+    mock_multiple_parquet_files,
+    backend,
+    file_type,
+    add_filename,
+    function_name,
+    cols_to_select,
+):
+    input_files = (
+        mock_multiple_jsonl_files
+        if file_type == "jsonl"
+        else mock_multiple_parquet_files
+    )
+    if function_name == "read_data_fpp":
+        func = read_data_fpp
+        read_kwargs = {"files_per_partition": 1}
+    elif function_name == "read_data_blocksize":
+        func = read_data_blocksize
+        read_kwargs = {"blocksize": "128Mib"}
+
+    df = func(
+        input_files=input_files,
+        backend=backend,
+        file_type=file_type,
+        add_filename=add_filename,
+        input_meta=None,
+        columns=list(cols_to_select) if cols_to_select else None,
+        **read_kwargs,
+    )
+    if not cols_to_select:
+        cols_to_select = ["id", "text"]
+
+    if not add_filename:
+        assert list(df.columns) == sorted(cols_to_select)
+    else:
+        assert list(df.columns) == sorted(cols_to_select + ["filename"])
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        "pandas",
+        pytest.param("cudf", marks=pytest.mark.gpu),
+    ],
+)
+@pytest.mark.parametrize("function_name", ["read_data_blocksize", "read_data_fpp"])
```

Comment:
Same as above, however I could change it to

```python

@paremetrize("read_kwargs", [
   {"files_per_partition": 1, "blocksize" : None}, 
   {"files_per_partition": None, "blocksize" : "128MiB"}
])
```

And then below instead of `func` I can call `read_data`

Reply:
But then the naming is still slightly ugly i.e `pandas-read_kwargs-1-input_meta-2` rather than `pandas-read_data_blocksize-input_meta-2`

Reply:
Oh I see, in that case yeah it's up to you. Thanks.